### PR TITLE
Add sudo parameter to checkcommand object

### PR DESCRIPTION
**Pull request by erikanderson**
Add sudo parameter to check_command object. Based on this answer:

 https://monitoring-portal.org/index.php?thread/33161-icinga2-sudo-und-command/&postID=215251#post215251

### CODE REVIEW

**Comment by erikanderson** [comment]
@arioch any thoughts on this?

**Comment by erikanderson** [comment]
Bump

**Comment by arioch** [comment]
Looking good so far.
Could you maybe add a little spec test to cover this change?

File: templates/object/checkcommand.conf.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@ object CheckCommand "<%= @object_checkcommandname %>" {
   import "<%= t -%>"
   <%- end -%>
   <%- if @command -%>
-  command = [ <% @command.each_with_index do |cmd, i| %><% if @cmd_path -%><%= @cmd_path -%> + <% end -%>"<%= cmd -%>"<%= ', ' if i < (@command.size - 1) %><% end %> ]
+  command = [ <% if @sudo %>"/usr/bin/sudo", <% end %><% @command.each_with_index do |cmd, i| %><% if @cmd_path -%><%= @cmd_path -%> + <% end -%>"<%= cmd -%>"<%= ', ' if i < (@command.size - 1) %><% end %> ]
```
Comment (zachfi) [inline templates/object/checkcommand.conf.erb]: How about the sudo command path be moved to a param on the class to override?  Maybe `$sudo_cmd` or some such?  Just a thought.

Comment (erikanderson) [inline templates/object/checkcommand.conf.erb]: Thats a good idea, will add that in. I will add spec test as well soon

**Comment by erikanderson** [comment]
Added spec test @arioch

**Comment by arioch** [comment]
Could you rebase (fixup) that into a single commit?
Final request, I promise. 😄

**Comment by erikanderson** [comment]
@arioch No problem, better to do it right the first time. Squashed commits and tests should pass

**Comment by arioch** [comment]
Excellent stuff. 👍 
Thanks! 

Merged upstream.

**Pull request closed without merging**